### PR TITLE
swap

### LINKED ISSUE #422: add `swap` for `url`
It looks like `url` is missing a `swap`. It could be more efficient than using `std::swap`.

### LINKED ISSUE #468: swap(v, v) is legal and required to work.
all our swaps have to handle `swap(v,v)`

### PR DESCRIPTION
**Pull request by alandefreitas**
fix #422, fix #468

### CODE REVIEW

File: include/boost/url/static_url.hpp [hunk 1]
```diff
@@ -69,6 +69,31 @@ class BOOST_SYMBOL_VISIBLE
...
+
+    /** Swap the contents.
+
+        Exchanges the contents of this static_url with another
+        static_url. The contents of the urls are swapped in
+        linear time with no possibility of exceptions.
+        All views, iterators and references are invalidated.
+
+        @par Complexity
+
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:81]: delete this line

File: include/boost/url/static_url.hpp [hunk 2]
```diff
@@ -69,6 +69,31 @@ class BOOST_SYMBOL_VISIBLE
...
+        Linear in @ref size() + `other.size()`
+
+        @par Exception Safety
+
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:85]: delete this blank line

File: include/boost/url/static_url.hpp [hunk 3]
```diff
@@ -69,6 +69,31 @@ class BOOST_SYMBOL_VISIBLE
...
         this->url_base::copy(u);
     }
+
+    /** Swap the contents.
+
+        Exchanges the contents of this static_url with another
+        static_url. The contents of the urls are swapped in
+        linear time with no possibility of exceptions.
+        All views, iterators and references are invalidated.
+
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:79]: Missing `@par Example`

File: include/boost/url/static_url.hpp [hunk 4]
```diff
@@ -69,6 +69,31 @@ class BOOST_SYMBOL_VISIBLE
...
+        @par Complexity
+
+        Linear in @ref size() + `other.size()`
+
+        @par Exception Safety
+
+        Throws nothing.
+
+        @param other The static_url to swap with
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:88]: You dont have to say "the static_url" just say "The object to swap with"

File: include/boost/url/static_url.hpp [hunk 5]
```diff
@@ -212,6 +237,41 @@ class static_url
...
+
+        Throws nothing
+
+        @param lhs The static_url to exchange.
+
+        @param rhs The static_url to exchange.
+
+        If `&lhs == &rhs`, this function call has no effect.
+
+        @see @ref static_url::swap
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:267]: see refs each have their own line

File: include/boost/url/static_url.hpp [hunk 6]
```diff
@@ -212,6 +237,41 @@ class static_url
...
+
+        @par Exception Safety
+
+        Throws nothing
+
+        @param lhs The static_url to exchange.
+
+        @param rhs The static_url to exchange.
+
+        If `&lhs == &rhs`, this function call has no effect.
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:265]: This statement should go up in the description

File: include/boost/url/url.hpp [hunk 7]
```diff
@@ -219,6 +219,67 @@ class BOOST_SYMBOL_VISIBLE url
...
+
+        @param rhs The url to exchange.
+
+        If `&lhs == &rhs`, this function call has no effect.
+
+        @see @ref url::swap
+    */
+    friend
+    void
+    swap(url& lhs, url& rhs) noexcept
```
Comment (vinniefalco) [inline include/boost/url/url.hpp:277]: When you swap there is no notion of left or right, they just swap that's why I try to write `v0` and `v1`

Comment (alandefreitas) [reply include/boost/url/url.hpp:277]: Oh... I was in doubt so I got this from boost.json. No problem.

File: include/boost/url/static_url.hpp [hunk 8]
```diff
@@ -70,6 +70,38 @@ class BOOST_SYMBOL_VISIBLE
     {
         this->url_base::copy(u);
     }
+
+    /** Swap the contents.
+
+        Exchanges the contents of this static_url with another
+        static_url. The contents of the urls are swapped in
+        linear time with no possibility of exceptions.
```
Comment (vinniefalco) [inline include/boost/url/static_url.hpp:78]: This should be part of the Complexity and Exception Safety sections

**Pull request closed without merging**